File: WealthWise.Shared/Models/FinancialGoal.cs
namespace WealthWise.Shared.Models;

public class FinancialGoal
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal TargetAmount { get; set; }
    public DateTime TargetDate { get; set; }
    public GoalPriority Priority { get; set; }
    public GoalStatus Status { get; set; }
}

public enum GoalPriority
{
    Low,
    Medium,
    High
}

public enum GoalStatus
{
    NotStarted,
    InProgress,
    Completed
}

File: WealthWise.AI/FinancialGoal.py
from typing import Optional
from datetime import datetime
import uuid
from pydantic import BaseModel, Field, ConfigDict

class FinancialGoal(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    
    Id: Optional[str] = Field(default_factory=lambda: str(uuid.uuid4()), alias="Id")
    Name: Optional[str] = Field('', alias="Name")
    Description: Optional[str] = Field('', alias="Description")
    TargetAmount: Optional[float] = Field(0.0, alias="TargetAmount")
    TargetDate: Optional[datetime] = Field(None, alias="TargetDate")
    Priority: Optional[str] = Field(None, alias="Priority")
    Status: Optional[str] = Field(None, alias="Status")